easybuild/easyblocks/p/pytorch.py

```diff
@@ -0,0 +1,230 @@
+        self.cfg['tests'] = tests
+
+    @staticmethod
+    def get_dependency_options_for_version(pytorch_version):
+        """
+        PyTorch can enable some functionality based on available software or use system software instead of a submodule
+        This returns EasyBuild names of that and the flag that should be used when the dependency is found
+
+        The result is a list of tuples (enable_flag, eb_name)
+        """
+        pytorch_version = LooseVersion(pytorch_version)
+
+        def is_version_ok(version_range):
+            """Return True if the PyTorch version to be installed matches the version_range"""
+            min_version, max_version = version_range.split(':')
+            result = True
+            if min_version and pytorch_version < LooseVersion(min_version):
+                result = False
+            if max_version and pytorch_version >= LooseVersion(max_version):
+                result = False
+            return result
+
+        available_libs = (
+            # Format: (PyTorch flag to enable, EB name, '<min version>:<exclusive max version>')
+            # Use `None` for the EB name if no known EC exists
+            ('USE_FFMPEG=1', 'FFmpeg', '1.0.0:'),
+            ('USE_GFLAGS=1', 'gflags', '1.0.0:'),
+            ('USE_GLOG=1', 'glog', '1.0.0:'),
+
+            # For system libs check CMakeLists.txt, below `if(USE_SYSTEM_LIBS)`, order kept here
+            # NCCL handled specially as other env variables are requires for it
+            ('USE_SYSTEM_CPUINFO=1', None, '1.6.0:'),
+            ('USE_SYSTEM_SLEEF=1', None, '1.6.0:'),
+            ('USE_SYSTEM_GLOO=1', None, '1.6.0:'),
+            ('BUILD_CUSTOM_PROTOBUF=0', 'protobuf', '1.2.0:'),
+            ('USE_SYSTEM_EIGEN_INSTALL=1', 'Eigen', '1.0.0:'),
+            ('USE_SYSTEM_FP16=1', None, '1.6.0:'),
+            ('USE_SYSTEM_PTHREADPOOL=1', None, '1.6.0:'),
+            ('USE_SYSTEM_PSIMD=1', None, '1.6.0:'),
+            ('USE_SYSTEM_FXDIV=1', None, '1.6.0:'),
+            ('USE_SYSTEM_BENCHMARK=1', None, '1.6.0:'),  # Google Benchmark
+            ('USE_SYSTEM_ONNX=1', None, '1.6.0:'),
+            ('USE_SYSTEM_XNNPACK=1', None, '1.6.0:'),
+        )
+        return [(enable_opt, dep_name) for enable_opt, dep_name, version_range in available_libs
+                if is_version_ok(version_range)]
+
+    def configure_step(self):
+        """Custom configure procedure for PyTorch."""
+        super(EB_PyTorch, self).configure_step()
+
+        options = ['PYTORCH_BUILD_VERSION=' + self.version, 'PYTORCH_BUILD_NUMBER=1']
+        # Restrict parallelism
+        options.append('MAX_JOBS=%(parallel)s')
+
+        # BLAS Interface
+        if get_software_root('imkl'):
+            options.append('BLAS=MKL')
+        else:
+            # This is what PyTorch defaults to if no MKL is found. Make this explicit here
+            options.append('BLAS=Eigen')
```

Comment:
Shouldn't this imply that `Eigen` is included as a dependency, so check the value of `get_software_root('Eigen')`?

Reply:
Just checked: They download eigen as a submodule and use that (or better: we do the download)
However we might be able to skip the download if we use Eigen as a dependency. Will try that. But for this it doesn't matter: It is either the system version or a silent fallback to submodule

Reply:
OK, fine then